core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -165,7 +191,9 @@ object ConsumerPerformance {
 
         if (currentTimeMillis - lastReportTime >= config.reportingInterval) {
           if (config.showDetailedStats)
-            printProgressMessage(0, bytesRead, lastBytesRead, messagesRead, lastMessagesRead, lastReportTime, currentTimeMillis, config.dateFormat)
+            printProgressMessage(0, bytesRead, lastBytesRead, messagesRead, lastMessagesRead,
+              lastReportTime, currentTimeMillis, config.dateFormat, Some(testStartTime), Some(joinTime.get), Some(joinTimeMsInSingleRound))
```

Comment:
Is `testStartTime` correct? Shouldn't this be the beginning of the interval?

Reply:
`testStartTime` means the exact time where the entire consumer perf tool begins. Beginning of the interval for each round is 6th parameter of `printProgressMessage`. Does it make sense?

Reply:
What I am not understanding is why we use the test start time when the other periodic statistics are computed based on the the interval start time. Shouldn't we be consistent?